components/clarity-lsp/src/common/backend.rs

```diff
@@ -156,15 +157,35 @@ pub async fn process_notification(
                                 )
                                 .await
                             }
-                        }?
-                        .contracts_settings
-                        .get(&contract_location)
-                        .ok_or(format!(
-                            "No Clarinet.toml is associated to the contract {}",
-                            &contract_location.get_file_name().unwrap_or_default()
-                        ))?
-                        .clone()
-                        .clarity_version
+                        }?;
+
+                        if let Some(contract_metadata) =
+                            manifest.contracts_settings.get(&contract_location)
+                        {
+                            contract_metadata.clarity_version
+                        } else {
+                            // Check if custom boot contract
+                            let contract_name = contract_location
+                                .to_path_buf()
+                                .file_stem()
+                                .and_then(|s| s.to_str())
+                                .unwrap_or_default()
+                                .to_string();
+
+                            if manifest
+                                .project
+                                .override_boot_contracts_source
+                                .contains_key(&contract_name)
+                            {
+                                // Custom boot contracts don't specify clarity versions
+                                ClarityVersion::latest()
```

Comment:
Can we use `get_boot_contract_epoch_and_clarity_version()` here?

Reply:
Yes! good call